src/cbc_sdk/platform/policy_ruleconfigs.py

```diff
@@ -272,3 +299,316 @@ def set_assignment_mode(self, mode):
+        default_rule['action'] = action
+        self.set_parameter("default_rule", default_rule)
+
+    @property
+    def rule_groups(self):
+        """
+        Returns the list of rule groups in this rule configuration.
+
+        Returns:
+            list[FirewallRuleGroup]: The list of rule groups.
+        """
+        if not self._rule_groups_valid:
+            rg_param = self.get_parameter("rule_groups")
+            if rg_param is not None:
+                self._rule_groups = [HostBasedFirewallRuleConfig.FirewallRuleGroup(self._cb, d) for d in rg_param]
+                for group in self._rule_groups:
+                    group._init_parent(self)
+            else:
+                self._rule_groups = []
+            self._rule_groups_valid = True
+        return self._rule_groups
+
+    def new_rule_group(self, name, description):
+        """
+        Creates a new FirewallRuleGroup object.
+
+        Args:
+            name (str): The name of the new rule group.
+            description (str): The description of the new rule group.
+
+        Returns:
+            FirewallRuleGroup: The new rule group object.  Add it to this rule configuration with append_rule_group.
+        """
+        return HostBasedFirewallRuleConfig.FirewallRuleGroup(self._cb, {"name": name, "description": description,
+                                                                        "rules": []})
+
+    def append_rule_group(self, rule_group):
+        """
+        Appends a rule group to the list of rule groups in the rule configuration.
+
+        Args:
+            rule_group (FirewallRuleGroup): The rule group to be added.
+        """
+        self.rule_groups.append(rule_group)
+        rule_group._parent = self
+        self._mark_changed()
+
+    def new_rule(self, name, action, direction, protocol, remote_ip):
+        """
+        Creates a new FirewallRule object.
+
+        Args:
+            name (str): The name for the new rule.
+            action (str): The action to be taken by this rule. Valid values are "ALLOW," "BLOCK," and "BLOCK_ALERT."
+            direction (str): The traffic direction this rule matches. Valid values are "IN," "OUT," and "BOTH."
+            protocol (str): The network protocol this rule matches. Valid values are "TCP" and "UDP."
+            remote_ip (str): The remote IP address this rule matches.
+
+        Returns:
+            FirewallRule: The new firewall rule. Append it to a rule group using the group's append_rule method.
+        """
+        if action not in ("ALLOW", "BLOCK", "BLOCK_ALERT"):
+            raise ApiError(f"invalid rule action: {action}")
+        if direction not in ("IN", "OUT", "BOTH"):
+            raise ApiError(f"invalid rule direction: {direction}")
+        if protocol not in ("TCP", "UDP"):
+            raise ApiError(f"invalid rule protocol: {protocol}")
+        return HostBasedFirewallRuleConfig.FirewallRule(self._cb, {"action": action, "application_path": "*",
```

Comment:
While I still think we'll need default values in case _nothing_ is specified for any given value, the use of `kwargs` to specify optional fields is a good idea and I'll want to implement that.

Reply:
This has actually been implemented.